cli/pkg/kctrl/cmd/app/init/vendir.go

```diff
@@ -0,0 +1,228 @@
+	err := v.printVendirFile()
+	if err != nil {
+		return err
+	}
+	err = v.runVendirSync()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VendirStep) printVendirFile() error {
+	vendirFileLocation := VendirFileName
+	v.ui.PrintActionableText(fmt.Sprintf("Printing %s", vendirFileLocation))
+	v.ui.PrintCmdExecutionText(fmt.Sprintf("cat %s", vendirFileLocation))
+	err := v.printFile(vendirFileLocation)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VendirStep) printFile(filePath string) error {
+	var stdoutBs, stderrBs bytes.Buffer
+
+	localCmdRunner := exec.NewPlainCmdRunner()
+	cmd := goexec.Command("cat", []string{filePath}...)
+	cmd.Stdin = nil
+	cmd.Stdout = &stdoutBs
+	cmd.Stderr = &stderrBs
+	localCmdRunner.Run(cmd)
+
+	result := exec.CmdRunResult{
+		Stdout: stdoutBs.String(),
+		Stderr: stderrBs.String(),
+	}
+	if result.Error != nil {
+		return fmt.Errorf("Printing file: %s", result.Stderr)
+	}
+
+	v.ui.PrintCmdExecutionOutput(result.Stdout)
+
+	return nil
+}
+
+// TODO vendir sync failure. Reproduce: In case of 429 from github, we dont show errors today.
+func (v *VendirStep) runVendirSync() error {
+	v.ui.PrintInformationalText("\nNext step is to run `vendir sync` to fetch the data from source to local directory. Vendir will sync the data into the upstream folder.")
+	v.ui.PrintActionableText("Running vendir sync")
+	v.ui.PrintCmdExecutionText("vendir sync -f vendir.yml")
+	var stdoutBs, stderrBs bytes.Buffer
+
+	localCmdRunner := exec.NewPlainCmdRunner()
+	cmd := goexec.Command("vendir", []string{"sync", "-f", VendirFileName}...)
+	cmd.Stdin = nil
+	cmd.Stdout = &stdoutBs
+	cmd.Stderr = &stderrBs
+	localCmdRunner.Run(cmd)
+
+	result := exec.CmdRunResult{
+		Stdout: stdoutBs.String(),
+		Stderr: stderrBs.String(),
+	}
+	if result.Error != nil {
+		return fmt.Errorf("Vendir sync failed. %s", result.Stderr)
+	}
+
+	v.ui.PrintInformationalText("\nTo validate that data has been fetched, lets list down the files")
+	v.ui.PrintActionableText(fmt.Sprintf("Validating by listing files"))
+	v.ui.PrintCmdExecutionText(fmt.Sprintf("ls -lR %s", VendirSyncDirectory))
+	err := v.listFiles(VendirSyncDirectory)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VendirStep) listFiles(dir string) error {
+	var stdoutBs, stderrBs bytes.Buffer
+
+	localCmdRunner := exec.NewPlainCmdRunner()
+	cmd := goexec.Command("ls", []string{"-lR", dir}...)
+	cmd.Stdin = nil
+	cmd.Stdout = &stdoutBs
+	cmd.Stderr = &stderrBs
+	localCmdRunner.Run(cmd)
+
+	result := exec.CmdRunResult{
+		Stdout: stdoutBs.String(),
+		Stderr: stderrBs.String(),
+	}
+	if result.Error != nil {
+		return fmt.Errorf("Listing files.\n %s", result.Stderr)
+	}
+	v.ui.PrintCmdExecutionOutput(result.Stdout)
+	return nil
```

Comment:
i dont think we need to list files especially if we take a dependence on ls binary.